lms/djangoapps/branding/__init__.py

```diff
@@ -17,12 +17,16 @@
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
 
-def get_visible_courses():
+def get_visible_courses(org=None):
     """
     Return the set of CourseDescriptors that should be visible in this branded instance
     """
-    filtered_by_org = microsite.get_value('course_org_filter')
-    courses = CourseOverview.get_all_courses(org=filtered_by_org)
+    microsite_org = microsite.get_value('course_org_filter')
+
+    orgs = [org, microsite_org]
+    orgs = [org for org in orgs if org is not None]
+
+    courses = CourseOverview.get_all_courses(orgs=orgs)
```

Comment:
"Microsite always wins" is intended to avoid accidentally exposing courses from non-microsite orgs if this function is used for filtering in the context of a microsite. My understanding is that `microsite_org` will be `None` for edx.org deployments.

I don't think querying against the intersection is what we want here. The `org` passed as an argument to this function will typically not be the same as the configured microsite org (e.g., the value returned by `microsite.get_value('course_org_filter')`). Querying against the intersection in this case would result in an empty orgs list and cause all visible course runs to be returned. Querying against the union fetches all applicable results, after which the filtering on line 42 &mdash; previously useless &mdash; now serves as a safety net ensuring that courses belonging to orgs other than the configured microsite org are not leaked.